harness/determined/deploy/local/cli.py

```diff
@@ -119,7 +113,7 @@ def deploy_local(args: argparse.Namespace) -> None:
                     Arg(
                         "--storage-host-path",
                         type=Path,
-                        default=DEFAULT_STORAGE_HOST_PATH,
```

Comment:
I changed the behavior.  If you provide `--storage-host-path`, it will override what is in the `master.yaml`.  This is normally how cli flags work with respect to a config file.

As a result, the default had to be `None` so that the application could distinguish between "--storage-host-path" not provided and "--storage-host-path was provided and was set to the default value".